ejdb 2.0.59 OSX build failure

👋 trying to build the latest release, but run into some build issue. The error log is as below:


<details>
<summary>build failure</summary>

```
[ 92%] Linking C static library libejdb2-2.a
cd /tmp/ejdb-20210321-51192-psjxin/ejdb-2.0.59/build/src && /usr/local/Cellar/cmake/3.19.7/bin/cmake -P CMakeFiles/ejdb2_s.dir/cmake_clean_target.cmake
cd /tmp/ejdb-20210321-51192-psjxin/ejdb-2.0.59/build/src && /usr/local/Cellar/cmake/3.19.7/bin/cmake -E cmake_link_script CMakeFiles/ejdb2_s.dir/link.txt --verbose=1
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip: no files specified
make[2]: *** [src/libejdb2.2.0.59.dylib] Error 1
make[2]: *** Deleting file `src/libejdb2.2.0.59.dylib'
make[1]: *** [src/CMakeFiles/ejdb2.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/usr/bin/ar qc libejdb2-2.a CMakeFiles/ejdb2_s.dir/ejdb2.c.o CMakeFiles/ejdb2_s.dir/util/lwre.c.o CMakeFiles/ejdb2_s.dir/util/utf8proc.c.o CMakeFiles/ejdb2_s.dir/jbl/binn.c.o CMakeFiles/ejdb2_s.dir/jbl/jbl.c.o CMakeFiles/ejdb2_s.dir/jbl/jbl_json.c.o CMakeFiles/ejdb2_s.dir/jql/jql.c.o CMakeFiles/ejdb2_s.dir/jql/jqp.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_consumer.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_dup_scanner.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_full_scanner.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_pk_scanner.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_selection.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_sorter_consumer.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_uniq_scanner.c.o CMakeFiles/ejdb2_s.dir/jbi/jbi_util.c.o CMakeFiles/ejdb2_s.dir/jbr/jbr.c.o
/usr/bin/ranlib libejdb2-2.a
[ 92%] Built target ejdb2_s
make: *** [all] Error 2
```

</details>


Full build log is in here, https://github.com/Homebrew/homebrew-core/runs/2160527827
relates to https://github.com/Homebrew/homebrew-core/pull/73617

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -22,6 +22,12 @@ if(APPLE)
       CACHE STRING "Directory to install frameworks to.")
 endif()
 
+if(APPLE)
+  set(STRIP_CMD strip -x)
+else()
+  set(STRIP_CMD strip -s)
+endif()
+
 include(CheckIncludeFile)
 include(CheckIncludeFiles)
 include(CheckLibraryExists)
@@ -286,7 +292,7 @@ if(BUILD_SHARED_LIBS)
     add_custom_command(
       TARGET ejdb2
       POST_BUILD
-      COMMAND strip -s $<TARGET_FILE:ejdb2>)
+      COMMAND ${STRIP_CMD} $<TARGET_FILE:ejdb2>)
   endif()
 
   set_target_properties(

--- a/src/bindings/ejdb2_dart/CMakeLists.txt
+++ b/src/bindings/ejdb2_dart/CMakeLists.txt
@@ -43,7 +43,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Release")
   add_custom_command(
     TARGET ejdb2_dart
     POST_BUILD
-    COMMAND strip -s $<TARGET_FILE:ejdb2_dart>)
+    COMMAND ${STRIP_CMD} $<TARGET_FILE:ejdb2_dart>)
 endif()
 
 set_target_properties(

--- a/src/bindings/ejdb2_jni/src/CMakeLists.txt
+++ b/src/bindings/ejdb2_jni/src/CMakeLists.txt
@@ -123,7 +123,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Release")
   add_custom_command(
     TARGET ejdb2jni
     POST_BUILD
-    COMMAND strip -s $<TARGET_FILE:ejdb2jni>)
+    COMMAND ${STRIP_CMD} $<TARGET_FILE:ejdb2jni>)
 endif()
 
 install(

--- a/src/bindings/ejdb2_node/CMakeLists.txt
+++ b/src/bindings/ejdb2_node/CMakeLists.txt
@@ -61,7 +61,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Release")
   add_custom_command(
     TARGET ejdb2_node
     POST_BUILD
-    COMMAND strip -s $<TARGET_FILE:ejdb2_node>)
+    COMMAND ${STRIP_CMD} $<TARGET_FILE:ejdb2_node>)
 endif()
 
 configure_file(package.json ${NODE_PUB_DIR}/package.json @ONLY)

--- a/src/jbs/CMakeLists.txt
+++ b/src/jbs/CMakeLists.txt
@@ -6,7 +6,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Release")
   add_custom_command(
     TARGET jbs
     POST_BUILD
-    COMMAND strip -s $<TARGET_FILE:jbs>)
+    COMMAND ${STRIP_CMD} $<TARGET_FILE:jbs>)
 endif()
 
 if(DO_INSTALL_CORE)

